skimage/morphology/grey.py

```diff
@@ -62,16 +130,15 @@ def erosion(image, selem=None, out=None, shift_x=False, shift_y=False):
            [0, 0, 0, 0, 0]], dtype=uint8)
 
     """
-
-    if image is out:
-        raise NotImplementedError("In-place erosion not supported!")
-    image = img_as_ubyte(image)
-    selem = img_as_ubyte(selem)
-    return cmorph._erode(image, selem, out=out,
-                         shift_x=shift_x, shift_y=shift_y)
+    selem = np.array(selem)
+    selem = _shift_selem(selem, shift_x, shift_y)
+    if out is None:
+        out = np.empty_like(image)
+    nd.grey_erosion(image, footprint=selem, output=out)
+    return out
 
 
-@default_fallback
+@default_selem
 def dilation(image, selem=None, out=None, shift_x=False, shift_y=False):
```

Comment:
@JDWarner I think you're quite right that an eccentric structuring element can always be made equivalent with a padded, centered one. There is a performance tradeoff, but it is negligible for the only case that we previously supported, where one axis had length 2. And at any rate, performance improvements should go to ndimage, not here, where it is redundant effort. Hopefully we'll get a good GSoC candidate to do the Cython rewrite this summer!